Src/AutoFixtureUnitTest/FixtureTest.cs

```diff
@@ -948,8 +948,8 @@ public void InjectCustomUpperLimitWillCauseSutToReturnNumbersInLimit()
             var result = sut.Create<DoublePropertyHolder<int, long>>();
             // Verify outcome
             Assert.True(
-                (result.Property1 >= lower && result.Property1 < upper) &&
-                (result.Property2 >= lower && result.Property2 < upper)
+                (result.Property1 >= lower && result.Property1 <= upper) &&
+                (result.Property2 >= lower && result.Property2 <= upper)

```

Comment:
Impressive analysis @Pvlerick and @jryanmoseley :+1: 

I must admit that even I, who created this class in the first place, couldn't answer all the questions without doing analysis :open_mouth:

---

As a side note, the fact that such an analysis is required makes me think of [this](http://fsharpforfunandprofit.com/posts/is-your-language-unreasonable/) article, and [this](http://blog.ploeh.dk/2015/04/13/less-is-more-language-features/), and [this](http://blog.ploeh.dk/2015/04/15/c-will-eventually-get-all-f-features-right/), and others as well :relieved: